PD: e2e tests on mac are flakey

# Overview

e2e tests on mac are flakey in PD. Let's start documenting error patterns here and then circle back to fix them.

List of failures:

2/17/2021: encountered an error during `make test e2e`: 

```
[protocol-designer-tests] 25l[15:21:52]  Verifying Cypress can run /Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app [started]
[protocol-designer-tests] [15:22:23]  Verifying Cypress can run /Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app [failed]
[protocol-designer-tests] 25hCypress verification timed out.
[protocol-designer-tests] 
[protocol-designer-tests] This command failed with the following output:
[protocol-designer-tests] 
[protocol-designer-tests] /Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/MacOS/Cypress --no-sandbox --smoke-test --ping=953
[protocol-designer-tests] 
[protocol-designer-tests] ----------
[protocol-designer-tests] 
[protocol-designer-tests] Command timed out after 30000 milliseconds: /Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/MacOS/Cypress --no-sandbox --smoke-test --ping=953
[protocol-designer-tests] Timed out
```

3/3/2021 failed on `newProtocolWithTC.spec.js`
```
[protocol-designer-tests]   1) Protocols with Modules builds a new protocol with mag deck designs protocol with thermocycler:
[protocol-designer-tests]      CypressError: Timed out retrying: cy.select() failed because it could not find a single <option> with value or text matching: 'THERMO TC Well'
```
3/3/2021 failed on `make setup-js`
```
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "https://registry.yarnpkg.com/rxjs/-/rxjs-6.5.3.tgz: ESOCKETTIMEDOUT".
info If you think this is a bug, please open a bug report with the information provided in "/Users/runner/work/opentrons/opentrons/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
info There appears to be trouble with your network connection. Retrying...
```

3/4/2021 failed after all tests passed due to a memory leak...?
```
[protocol-designer-tests] /Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/MacOS/Cypress[4351]: ../../third_party/electron_node/src/node_platform.cc:444:std::shared_ptr<PerIsolatePlatformData> node::NodePlatform::ForIsolate(v8::Isolate *): Assertion `data' failed.
[protocol-designer-tests]  1: 0x107433f85 node::Abort() [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  2: 0x107433d0f node::Assert(node::AssertionInfo const&) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  3: 0x10749b407 node::options_parser::SplitHostPort(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >*) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  4: 0x1029b5aaf v8::internal::Isolate::SetUseCounterCallback(void (*)(v8::Isolate*, v8::Isolate::UseCounterFeature)) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  5: 0x1029f89b8 v8::internal::IncrementalMarking::V8Step(double, v8::internal::IncrementalMarking::CompletionAction, v8::internal::StepOrigin) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  6: 0x1029eb3aa v8::internal::IncrementalMarking::AdvanceOnAllocation() [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  7: 0x1029eb03e v8::internal::IncrementalMarking::Observer::Step(int, unsigned long, unsigned long) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  8: 0x1027d3e3b v8::internal::Heap::KeepDuringJob(v8::internal::Handle<v8::internal::JSReceiver>) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests]  9: 0x102a337e2 v8::internal::NewSpace::EnsureAllocation(int, v8::internal::AllocationAlignment) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests] 10: 0x1027ce30b v8::internal::Heap::PromotedExternalMemorySize() [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests] 11: 0x1027ce111 v8::internal::Heap::PromotedExternalMemorySize() [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests] 12: 0x1027ce511 v8::internal::Heap::PromotedExternalMemorySize() [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests] 13: 0x1029d038a v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests] 14: 0x102c59adf v8::internal::RegExp::CompileForTesting(v8::internal::Isolate*, v8::internal::Zone*, v8::internal::RegExpCompileData*, v8::base::Flags<v8::internal::JSRegExp::Flag, int>, v8::internal::Handle<v8::internal::String>, v8::internal::Handle<v8::internal::String>, bool) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests] 15: 0x10329b499 v8::internal::compiler::ZoneStats::ReturnZone(v8::internal::Zone*) [/Users/runner/Library/Caches/Cypress/4.0.2/Cypress.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
[protocol-designer-tests] The Test Runner unexpectedly exited via a exit event with signal SIGABRT
[protocol-designer-tests] 
[protocol-designer-tests] Please search Cypress documentation for possible solutions:
```
3/9/2021 failed because cypress could not find a stepitem data id
```
[protocol-designer-tests]   2) Protocols with Modules
[protocol-designer-tests]        build a new protocol with mag deck
[protocol-designer-tests]          deletes and replaces modules:
[protocol-designer-tests]      AssertionError: Timed out retrying after 4000ms: Expected to find element: `[data-test="StepItem_4"]`, but never found it.
```

--- a/protocol-designer/cypress/integration/newProtocolWithModules.spec.js
+++ b/protocol-designer/cypress/integration/newProtocolWithModules.spec.js
@@ -13,6 +13,9 @@ const moduleModelDropdown =
 const slotOne = 'foreignObject[x="0"][y="0"]'
 const slotThree = 'foreignObject[x="265"][y="0"]'
 const slotFive = 'foreignObject[x="132.5"][y="90.5"]'
+const tempDeckBlockWithExtraSpace = 'Temp Deck Block '
+// Note: reason this is needed is because of a cypress race condition. see `https://stackoverflow.com/questions/66257661/how-to-resolve-missing-characters-bug-in-cypress-type-command`
+const magenticDeckWellWithExtraSpace = 'Mag Deck Well '
 
 describe('Protocols with Modules', () => {
   beforeEach(() => {
@@ -167,7 +170,7 @@ describe('Protocols with Modules', () => {
         })
       })
       cy.get('input[class*="LabwareOverlays__name_input"]').type(
-        'Mag Deck Well '
+        magenticDeckWellWithExtraSpace
       )
 
       // Add labware to Temp Deck
@@ -193,7 +196,7 @@ describe('Protocols with Modules', () => {
       })
       cy.contains(aluminumWellBlock, { matchCase: false })
       cy.get('input[class*="LabwareOverlays__name_input"]').type(
-        'Temp Deck Block '
+        tempDeckBlockWithExtraSpace
       )
 
       // Add first liquid

--- a/protocol-designer/cypress/integration/newProtocolWithModules.spec.js
+++ b/protocol-designer/cypress/integration/newProtocolWithModules.spec.js
@@ -167,7 +167,7 @@ describe('Protocols with Modules', () => {
         })
       })
       cy.get('input[class*="LabwareOverlays__name_input"]').type(
-        'Mag Deck Well'
+        'Mag Deck Well '
       )
 
       // Add labware to Temp Deck
@@ -193,7 +193,7 @@ describe('Protocols with Modules', () => {
       })
       cy.contains(aluminumWellBlock, { matchCase: false })
       cy.get('input[class*="LabwareOverlays__name_input"]').type(
-        'Temp Deck Block'
+        'Temp Deck Block '
       )
 
       // Add first liquid

--- a/protocol-designer/cypress/integration/newProtocolWithTC.spec.js
+++ b/protocol-designer/cypress/integration/newProtocolWithTC.spec.js
@@ -26,6 +26,8 @@ const moduleModelDropdown =
 const thermocyclerSlot = 'foreignObject[x="12"][y="267"]'
 const slotSeven = 'foreignObject[x="0"][y="181"]'
 
+const thermocyclerWellBlockWithExtraSpace = 'TC Well ' // Note: reason this is needed is because of a cypress race condition. see `https://stackoverflow.com/questions/66257661/how-to-resolve-missing-characters-bug-in-cypress-type-command`
+
 describe('Protocols with Modules', () => {
   beforeEach(() => {
     cy.viewport('macbook-15')
@@ -142,7 +144,9 @@ describe('Protocols with Modules', () => {
           cy.contains('Nest 96 Well Plate', { matchCase: false }).click()
         })
       })
-      cy.get('input[class*="LabwareOverlays__name_input"]').type('TC Well ')
+      cy.get('input[class*="LabwareOverlays__name_input"]').type(
+        thermocyclerWellBlockWithExtraSpace
+      )
 
       // Add Thermocycler State Step
       cy.addStep('thermocycler')
@@ -268,7 +272,9 @@ describe('Protocols with Modules', () => {
             .click()
         })
       })
-      cy.get('input[class*="LabwareOverlays__name_input"]').type('TC Well ')
+      cy.get('input[class*="LabwareOverlays__name_input"]').type(
+        thermocyclerWellBlockWithExtraSpace
+      )
 
       // Add Thermocycler Profile Step
       cy.addStep('thermocycler')

--- a/protocol-designer/cypress/integration/newProtocolWithTC.spec.js
+++ b/protocol-designer/cypress/integration/newProtocolWithTC.spec.js
@@ -142,9 +142,7 @@ describe('Protocols with Modules', () => {
           cy.contains('Nest 96 Well Plate', { matchCase: false }).click()
         })
       })
-      cy.get('input[class*="LabwareOverlays__name_input"]').type(
-        thermocyclerWellBlock
-      )
+      cy.get('input[class*="LabwareOverlays__name_input"]').type('TC Well ')
 
       // Add Thermocycler State Step
       cy.addStep('thermocycler')
@@ -270,9 +268,7 @@ describe('Protocols with Modules', () => {
             .click()
         })
       })
-      cy.get('input[class*="LabwareOverlays__name_input"]').type(
-        thermocyclerWellBlock
-      )
+      cy.get('input[class*="LabwareOverlays__name_input"]').type('TC Well ')
 
       // Add Thermocycler Profile Step
       cy.addStep('thermocycler')

